internal/provider/provider.go

```diff
@@ -146,12 +158,16 @@ func (p *CiscoMerakiProvider) Configure(ctx context.Context, req provider.Config
 
 	// MERAKI BASE URL
 	if !data.BaseUrl.IsNull() {
```

Comment:
I think it's better to use the schema to populate default values rather than to throw an error, but since the underlying API client does look for env vars we could do a check for those when the terraform values are null, otherwise we fallback on the default values.